src/SimulationService.jl

```diff
@@ -1,77 +1,389 @@
+function start!(; host=SIMSERVICE_HOST, port=SIMSERVICE_PORT, kw...)
+    Threads.nthreads() > 1 || error("Server require `Thread.nthreads() > 1`.  Start Julia via `julia --threads=auto`.")
+    stop!()  # Stop server if it's already running
+    server_url[] = "http://$host:$port"
+    JobSchedulers.scheduler_start()
+    JobSchedulers.set_scheduler(max_cpu=0.5, max_mem=0.5, update_second=0.05, max_job=5000)
+    Oxygen.resetstate()
+    Oxygen.@get "/" health
+    Oxygen.@post "/{operation_name}" operation
+    Oxygen.@get "/jobs/status/{job_id}" job_status
+    Oxygen.@get "/jobs/result/{job_id}" job_result
+    Oxygen.@post "/jobs/kill/{job_id}" job_kill
+    Oxygen.serveparallel(; host, port, async=true, kw...)
+end
+
+#-----------------------------------------------------------------------------# stop!
 function stop!()
-    scheduler_stop()
-    terminate()
+    JobSchedulers.scheduler_stop()
+    Oxygen.terminate()
+end
+
+#-----------------------------------------------------------------------------# settings
+# Server
+SIMSERVICE_HOST                 = get(ENV, "SIMSERVICE_HOST", "0.0.0.0")
+SIMSERVICE_PORT                 = parse(Int, get(ENV, "SIMSERVICE_PORT", "8000"))
+
+# RabbitMQ (host=localhost)
+SIMSERVICE_RABBITMQ_ENABLED     = get(ENV, "SIMSERVICE_RABBITMQ_ENABLED", "false") == "true"
+SIMSERVICE_RABBITMQ_LOGIN       = get(ENV, "SIMSERVICE_RABBITMQ_LOGIN", "guest")
+SIMSERVICE_RABBITMQ_PASSWORD    = get(ENV, "SIMSERVICE_RABBITMQ_PASSWORD", "guest")
+SIMSERVICE_RABBITMQ_ROUTE       = get(ENV, "SIMSERVICE_RABBITMQ_ROUTE", "terarium")
+SIMSERVICE_RABBITMQ_PORT        = parse(Int, get(ENV, "SIMSERVICE_RABBITMQ_PORT", "5672"))
+
+# Terrarium Data Service (TDS)
+SIMSERVICE_ENABLE_TDS           = get(ENV, "SIMSERVICE_ENABLE_TDS", "true") == "true"
+SIMSERVICE_TDS_URL              = get(ENV, "SIMSERVICE_TDS_URL", "http://localhost:8001")
+SIMSERVICE_TDS_RETRIES   = parse(Int, get(ENV, "SIMSERVICE_TDS_RETRIES", "10"))
+
+
+#-----------------------------------------------------------------------------# utils
+JSON_HEADER = ["Content-Type" => "application/json"]
+
+get_json(url::String)::JSON3.Object = JSON3.read(HTTP.get(url, JSON_HEADER).body)
+
+# TODO: more tests.  This is an important function.
+function ode_system_from_amr(obj::JSON3.Object)
+    model = obj.model
+    ode = obj.semantics.ode
+
+    t = only(@variables t)
+    D = Differential(t)
+
+    statenames = [Symbol(s.id) for s in model.states]
+    statevars  = [only(@variables $s) for s in statenames]
+    statefuncs = [only(@variables $s(t)) for s in statenames]
+
+    # get parameter values and state initial values
+    paramnames = [Symbol(x.id) for x in ode.parameters]
+    paramvars = [only(@parameters $x) for x in paramnames]
+    paramvals = [x.value for x in ode.parameters]
+    sym_defs = paramvars .=> paramvals
+    initial_exprs = [MathML.parse_str(x.expression_mathml) for x in ode.initials]
```

Comment:
@jpfairbanks this is where we use the MathML, but we could just symbolics parse the other expressions.